File: packages/libdivecomputer_plugin/macos/Classes/LibdivecomputerPlugin.swift
import FlutterMacOS

public class LibdivecomputerPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        let messenger = registrar.messenger
        let api = DiveComputerHostApiImpl(messenger: messenger)
        DiveComputerHostApiSetup.setUp(binaryMessenger: messenger, api: api)
    }
}

File: packages/libdivecomputer_plugin/android/src/main/kotlin/com/submersion/libdivecomputer/LibdivecomputerPlugin.kt
package com.submersion.libdivecomputer

import io.flutter.embedding.engine.plugins.FlutterPlugin

class LibdivecomputerPlugin : FlutterPlugin {
    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        val context = binding.applicationContext
        val messenger = binding.binaryMessenger
        val api = DiveComputerHostApiImpl(context, messenger)
        DiveComputerHostApi.setUp(messenger, api)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        DiveComputerHostApi.setUp(binding.binaryMessenger, null)
    }
}
